cpp/src/centrality/betweenness_centrality.cu

```diff
@@ -26,10 +26,287 @@
 
 #include <gunrock/gunrock.h>
 
+#include "betweenness_centrality.cuh"
+
 namespace cugraph {
+namespace detail {
+
+template <typename VT, typename ET, typename WT, typename result_t>
+void BC<VT, ET, WT, result_t>::setup()
+{
+  // --- Set up parameters from graph adjList ---
+  number_of_vertices = graph.number_of_vertices;
+  number_of_edges    = graph.number_of_edges;
+  offsets_ptr        = graph.offsets;
+  indices_ptr        = graph.indices;
+}
+
+template <typename VT, typename ET, typename WT, typename result_t>
+void BC<VT, ET, WT, result_t>::configure(result_t *_betweenness,
+                                         bool _normalized,
+                                         bool _endpoints,
+                                         WT const *_weights,
+                                         VT const *_sources,
+                                         VT _number_of_sources)
+{
+  // --- Bind betweenness output vector to internal ---
+  betweenness       = _betweenness;
+  normalized        = _normalized;
+  endpoints         = _endpoints;
+  sources           = _sources;
+  number_of_sources = _number_of_sources;
+  edge_weights_ptr  = _weights;
+
+  // --- Working data allocation ---
+  ALLOC_TRY(&distances, number_of_vertices * sizeof(VT), nullptr);
```

Comment:
You should add the following device_vectors as private members of class `BC` in `betweenness_centrality.cuh`
```
rmm::device_vector<VT> distance_vec;
rmm::device_vector<VT> predecessor_vec;
rmm::device_vector<double> counter_vec;
rmm::device_vector<double> delta_vec;
```
In this configure function you should do the folllowing :

```
distance_vec.resize(number_of_vertices);
predecessor_vec.resize(number_of_vertices);
counter_vec.resize(number_of_vertices);
delta_vec.resize(number_of_vertices);
distances = distance_vec.data().get();
predecessors = predecessor_vec.data().get();
sp_counters = counter_vec.data().get();
deltas = delta_vec.data().get();
```

This will elimiate `ALLOC_TRY` and `ALLOC_FREE_TRY` from your code with minimal changes. Making the device_vectors private members ensures that they are alive for the lifetime of the object of class BC and are only destroyed along with BC. So make sure that the pointers you get from the allocation (`obj.data().get()`) are only used internally in the class.

Reply:
Every memory allocation that is being done to temporarily hold a value to be used internally by the code should be done via device_vectors or device_buffers this way.

Reply:
Changed the BC class accordingly in 4bdf822.